src/coreclr/jit/lsra.cpp

```diff
@@ -887,6 +887,17 @@ void LinearScan::setBlockSequence()
             }
         }
 
+#if defined(FEATURE_EH_FUNCLETS) && defined(TARGET_ARM)
```

Comment:
This code seems odd to me, since we're already handling the callfinally/always pair stuff in the loop above, for all platforms. But the code above is weird because the condition is loop-invariant. It seems like we should just hoist:
```
            // We treat BBCallAlwaysPairTail blocks as having EH flow, since we can't
            // insert resolution moves into those blocks.
            if (block->isBBCallAlwaysPairTail())
            {
                blockInfo[block->bbNum].hasEHBoundaryIn  = true;
                blockInfo[block->bbNum].hasEHBoundaryOut = true;
            }
```

above the `pred` loop (for all platforms), and then replace it in the loop with:
```
if (!block->isBBCallAlwaysPairTail())
   ...
```

Reply:
That's a good point. Fixed.